wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,368 @@
+              dplyr::group_by(lab_wwtp_unique_id, date) |>
+              dplyr::summarize(n_obs = dplyr::n()) |>
+              dplyr::ungroup() |>
+              dplyr::summarize(
+                n_duplicates = sum(n_obs > 1)
+              ) |>
+              dplyr::pull(n_duplicates)
+
+            this_ww_metadata <- tibble::tibble(
+              forecast_date = this_forecast_date,
+              location = this_location,
+              ww_data_present = 1,
+              n_sites,
+              n_labs,
+              pop_coverage,
+              state_pop,
+              avg_latency,
+              avg_sampling_freq,
+              n_duplicate_obs
+            )
+          } else {
+            this_ww_metadata <- tibble::tibble(
+              forecast_date = this_forecast_date,
+              location = this_location,
+              ww_data_present = 0,
+              n_sites = NA,
+              n_labs = NA,
+              pop_coverage = NA,
+              state_pop = state_pop,
+              avg_latency = NA,
+              avg_sampling_freq = NA,
+              n_duplicate_obs = NA
+            )
+          }
+
+
+          ww_metadata <- rbind(ww_metadata, this_ww_metadata)
+        },
+        error = function(e) {}
+      )
+    } else {
+      warning(glue::glue(
+        "File missing for {this_scenario}",
+        "in {this_location} on {this_forecast_date}"
+      ))
+      # Create a tibble of the combos that are missing, to save
+      this_failed_output <- tibble(
+        scenario = this_scenario,
+        location = this_location,
+        forecast_date = this_forecast_date
+      )
+      flag_failed_output <- rbind(flag_failed_output, this_failed_output)
+    }
+  }
+
+  if (nrow(flag_failed_output) != 0) {
+    # Save the missing files in a new subfolder in the eval_output_subdir
+    cfaforecastrenewalww::create_dir(file.path(
+      eval_output_subdir,
+      "files_missing", model_type
+    ))
+
+    write.csv(
+      flag_failed_output,
+      file.path(
+        eval_output_subdir, "files_missing", model_type,
+        glue::glue("ww_data_metadata.csv")
+      )
+    )
+  }
+
+  return(ww_metadata)
+}
+
+
+#' Get additional wastewater metadata
+#'
+#' @param granular_ww_metadata a tibble with a row for each forecast date
+#' location and columns that provide summaries of the wastewater data. This
+#' is focused on input data
+#' @param ww_forecast_date_locs_to_excl table of forecast date-locations
+#' to manually exclude
+#' @param convergence_df table containing convergence flags for every forecast
+#' date location combination
+#' @param table_of_loc_dates_w_ww table containing wastewater metadata
+#' for every location-forecast date with wastewater
+#'
+#' @return a tibble with a number of additional columns indicating whether
+#' or not there were any flags for manual exclusions, convergence issues,
+#' or wastewater quality issues
+#' @export
+get_add_ww_metadata <- function(granular_ww_metadata,
```

Comment:
This function name is a bit opaque, as is the argument name `granualar_ww_metadata`).

It seems like the key thing that's actually going on here is joining wastewater data quality metrics (which are based on input data) to run convergence flags (which are based on mcmc output).

The purpose of the joins with `ww_forecast_date_locs_to_excl` and `table_of_loc_dates_w_ww` is currently also a bit opaque in their purpose. Is the aim to add a "was this manually excluded" column and a "was there sufficient wastewater on this date" column, respectively?

Reply:
Yeah so `get_add_ww_metadata()` just joins the other wastewater exclusions to the one row per forecast-date-location table (`granular_ww_metadata`)

Reply:
And yeah, its to be able to slice and dice across all of those reasons and get summaries

Reply:
`get_granular_ww_metdata()` makes a large table with one row per location-foercast-date with a bunch of metadata for that particular wastewater data